js/src/lib/components/DropdownEntry/DropdownEntry.svelte

```diff
@@ -10,25 +10,40 @@
 	export let underline = false;
 	export let onClick: (e: MouseEvent) => void = () => {};
 	export let targetBlank = false;
+	export let type: "link" | "button" | "submit" | undefined = undefined;
+
+	const element = type === undefined
+		? href !== undefined ? "a" : "button"
+		: type === "link" ? "a" : "button";
 </script>
 
 <li>
-	<a
-		class="flex items-center hover:bg-gray-50 dark:hover:bg-gray-800 cursor-pointer px-3 py-1.5 whitespace-nowrap 
-			{classNames} {underline ? 'hover:underline' : ''}"
+	<svelte:element
+		this={element}
+		class="flex items-center w-full text-left hover:bg-gray-50 dark:hover:bg-gray-800 cursor-pointer px-3 py-1.5 whitespace-nowrap
+		{classNames} {underline ? 'hover:underline' : ''}"
 		{href}
-		on:click={onClick}
-		rel={noFollow ? "nofollow" : undefined}
-		target={targetBlank ? "_blank" : undefined}
+		rel={element === "a" && noFollow ? "nofollow" : undefined}
+		target={element === "a" && targetBlank ? "_blank" : undefined}
+		on:click={(e) => {
+			if (type === "submit") {
+				e.stopPropagation();
+			}
+			onClick(e);
```

Comment:
It would be better to forward a native event but probably out of scope of this PR (as we try to remove this callback pattern everywhere).

Reply:
I was also tempted to move it to native events...